packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -379,14 +375,12 @@ class ProductVariantInventoryService extends TransactionBaseService {
     return await Promise.all(
       variantInventory.map(async (inventoryPart) => {
         const itemQuantity = inventoryPart.required_quantity * quantity
-        return await this.inventoryService_
-          .withTransaction(this.activeManager_)
-          .createReservationItem({
-            ...toReserve,
-            location_id: locationId as string,
-            inventory_item_id: inventoryPart.inventory_item_id,
-            quantity: itemQuantity,
-          })
+        return await this.inventoryService_.createReservationItem({
+          ...toReserve,
+          location_id: locationId as string,
+          inventory_item_id: inventoryPart.inventory_item_id,
+          quantity: itemQuantity,
+        })
```

Comment:
@carlos-r-l-rodrigues, @adrien2p - in a place like this i guess we will need an orchestrator to take care of business - how will this work in a loop?

Reply:
I talked to @adrien2p earlier and it sounded like we atm don't need it since we don't to cross-module transactions, however I might be wrong and as long as this throws it should be fine since we then cancel the core transaction, however I would love to get @carlos-r-l-rodrigues take on this

Reply:
In this particular case, as we have a loop and the `createReservationItem` can fail in some calls but succeed in others we need to handle the compensation for that.

The variant creation, while creating a product, is a similar case. I'm applying the [TO there](https://github.com/medusajs/medusa/blob/ad99a8bd24567fa8ed96af519ad731da8e24b174/packages/medusa/src/api/routes/admin/products/create-product.ts#L188
).

I believe this pattern will be common, and it would be nice if we support bulk operations for methods like this, then when there isn't multiple modules being called the TO use could be dropped.

Reply:
In the case of such mutations, it will require a TO indeed. The other case was a read @pKorsholm 😀 in the case of a loop you can have a look at the product update end point hander. We basically store the transactions in case of later rollback

Reply:
Instead of a loop, if the operation become bulk the rollback will do it as a bulk as well since the bulk create step will probably return something either the ids or entities

Reply:
alright @srindom how do you wanna move forward. I would suggest we merge this pr as it still doesn't work in develop and create tickets to fix in subsequent prs, wdyt?